### ISSUE TITLE
Package installed from private repository is being listed as being installed from public repository

### ISSUE THREAD

**Issue by Smith8154** [microsoft/winget-cli#4064]
### Brief description of your issue

When installing an application from our internal WinGet repository, WinGet is listing the package as being installed from the public WinGet repository.

### Steps to reproduce

Upload a package to an internal WinGet repository, where the same package exists in the public WinGet repository.
Install the package from the private repository.
Export the list of installed packages. It will show the package as being installed from the public repository, not the private one.

### Expected behavior

I would expect WinGet to show the package as being installed from the correct repository.

### Actual behavior

WinGet shows the package as being installed from the wrong repository.

### Environment

```shell
Windows Package Manager v1.6.3482
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19045.3803
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.21.3482.0
```

**Comment by Smith8154** [comment]
This doesn't seem to be an issue on every machine. I just pushed the package to 35 machine and it is reporting incorrectly on about half of them.

**Comment by denelon** [comment]
Was this application installed on any of the devices before the push?

It's possible an entry from the installed database showed winget as the source before the private source was added.

Possible work around:
If you can test, try uninstalling the app and then reinstalling from the private source.

We don't have a notion of an order of preference for a source when the package is available in more than one source yet.

**Comment by Smith8154** [comment]
Yes, the application was already installed. I tried to upgrade it to a newer version.

Unfortunately, uninstalling it and reinstalling did not work. I even updated the package to have the same ID as the public repository to see if that would help, but it did not. Oddly enough, this was not an issue with the previous version.

In doing this testing, I may have found why I am having issues. I am having trouble with the internal repository after I uploaded a new version of an (unrelated) application. It looks like that may be keeping WinGet from exporting packages from the private repository. This is the error I am seeing on the other repository:

```
2024-01-12 01:06:16.583 [REPO] Finding available package from installed package using system reference search: Query:[none] Include:ProductCode='sapgui'[Exact] Include:NormalizedNameAndPublisher='sapguiforwindows'+'sapse'[Exact] Include:NormalizedNameAndPublisher='sapguiforwindows(x86)'+'sapse'[Exact]
2024-01-12 01:06:16.586 [REPO] Sending http GET request to: https://internal-repo-url.net/api/packageManifests/Company.SAPGUI
2024-01-12 01:06:16.690 [REPO] Response status: 200
2024-01-12 01:06:16.691 [FAIL] C:\__w\1\s\external\pkg\src\AppInstallerSharedLib\Versions.cpp(573)\WindowsPackageManager.dll!00007FF888AF6324: (caller: 00007FF888AF8E05) Exception(1) tid(727c) 80070057 The parameter is incorrect.
```

**Comment by denelon** [comment]
@Smith8154 I'm going to transfer this issue to the winget-cli-restsource repository so we can look into troubleshooting what went wrong since it seems like it may be an issue with the REST implementation then.
